src/Core/Domain/Customer/Query/SearchCustomers.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Customer\Query;
+
+/**
+ * Searchers for customers by query matching on first name, last name, email and id
+ */
+class SearchCustomers
```

Comment:
i dont feel its the right name for `Query`, any suggestions?

Reply:
Why not ?
I googled "search query CQRS" to check, found 2 top examples with Search and Query in the class name so it looks fine.
https://blog.shadowhand.me/cqrs-search-queries/
https://www.reddit.com/r/PHP/comments/5b18jc/cqrs_search_queries/